FindAIETools.cmake

```diff
@@ -0,0 +1,218 @@
+#
+#  3.  Neither the name of the copyright holder nor the names of its 
+#      contributors may be used to endorse or promote products derived from 
+#      this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, 
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR 
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR 
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, 
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, 
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+#  OR BUSINESS INTERRUPTION). HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+#  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR 
+#  OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF 
+#  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+###############################################################################
+
+###########################################################
+#
+## 1: Setup:
+# The following environmental variables are optionally searched for defaults:
+#  none.
+#
+## 2: Variables
+# The following are set after configuration is done: 
+#  
+#  AIETOOLS_DIR - The path to AIETools installation directory for the specified version
+#  AIETOOLS_XCHESSCC - 'xchesscc' with full path
+#  AIETOOLS_XCHESS_MAKE - 'xchessmk' with full path
+#
+## 3: Components
+# The following components are supported:
+###  3.1. AIE
+#  AIETOOLS_AIE_INCLUDE_DIR - AIE full include path
+#  AIETOOLS_AIE_LIBME - 'libme.a' with full path
+#  AIETOOLS_AIE_LIBC - 'libc.a' with full path
+#  AIETOOLS_AIE_LIBM - 'libm.a' with full path
+#  AIETOOLS_AIE_RUNTIME_INCLUDE_DIR - AIE runtime full include path
+#  AIETOOLS_AIE_LIBSOFTFLOAT - 'softfloat.a' with full path
+#
+### 3.2. AIE2
+#  AIETOOLS_AIE2_INCLUDE_DIR - AIE2 full include path
+#  AIETOOLS_AIE2_LIBME - 'libme.a' with full path
+#  AIETOOLS_AIE2_LIBC - 'libc.a' with full path
+#  AIETOOLS_AIE2_LIBM - 'libm.a' with full path
+#  AIETOOLS_AIE2_RUNTIME_INCLUDE_DIR - AIE2 runtime full include path
+#  AIETOOLS_AIE2_LIBSOFTFLOAT - 'softfloat.a' with full path
+#----------------------------------------------------------
+
+include(FindPackageHandleStandardArgs)
+
+cmake_policy(SET CMP0144 NEW)
+
+list(LENGTH AIETools_FIND_COMPONENTS componentsSize)
+if(componentsSize EQUAL 0)
+ 	message(STATUS "Error: AIE version needed as COMPONENT, setting AIE as default")
+	list(APPEND AIETools_FIND_COMPONENTS "AIE")
+endif()
+
+#if Xilinx tools correctly installed they are added to $ENV{PATH} one of CMake's default search paths
+
+# Find aiecompiler to derive version
+find_program(AIETOOLS_AIECOMPILER aiecompiler)
+if(NOT AIETOOLS_AIECOMPILER)
+	message(STATUS "Unable to find aiecompiler")
+else(NOT AIETOOLS_AIECOMPILER)
```

Comment:
I'm guessing this `else` with a condition inside is a typo and doesn't do anything?
@denolf